File: src/java/foundgine-runtime/src/main/java/com/foundgine/runtime/PlanApproval.java
package com.foundgine.runtime;

import com.foundgine.core.semantic.SemanticRequest;

import java.time.Instant;

/**
 * Port of {@code Foundgine.Runtime.PlanApproval}.
 *
 * <p>Approval bound to the exact authorized semantic plan represented by the
 * fingerprint. It is not an authorization grant and cannot be reused for a
 * different plan.
 *
 * <p>C#'s {@code DateTimeOffset} is ported as {@link Instant}, matching this
 * port's established convention.
 */
public record PlanApproval(
        SemanticRequest request,
        String approvalId,
        String planFingerprint,
        String semanticModelVersion,
        int capabilityContractVersion,
        int capabilityVersion,
        int intentVersion,
        int planVersion,
        String approvedBy,
        Instant approvedAt) {
}

File: src/csharp/Foundgine.Runtime/Api/PlanApproval.cs
using Foundgine.Core.Semantic;

namespace Foundgine.Runtime;

/// <summary>
/// Approval bound to the exact authorized semantic plan represented by the
/// fingerprint. It is not an authorization grant and cannot be reused for a
/// different plan.
/// </summary>
public sealed record PlanApproval(
    SemanticRequest Request,
    string ApprovalId,
    string PlanFingerprint,
    string SemanticModelVersion,
    int CapabilityContractVersion,
    int CapabilityVersion,
    int IntentVersion,
    int PlanVersion,
    string ApprovedBy,
    DateTimeOffset ApprovedAt);